src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -8,55 +8,76 @@
 package frc.robot.subsystems;
 
 import com.ctre.phoenix6.hardware.TalonFX;
+import com.nrg948.preferences.RobotPreferences;
+import com.nrg948.preferences.RobotPreferencesLayout;
+import com.nrg948.preferences.RobotPreferencesValue;
 import edu.wpi.first.math.controller.ElevatorFeedforward;
 import edu.wpi.first.math.controller.ProfiledPIDController;
 import edu.wpi.first.math.system.plant.DCMotor;
 import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.networktables.GenericEntry;
 import edu.wpi.first.util.datalog.BooleanLogEntry;
 import edu.wpi.first.util.datalog.DoubleLogEntry;
 import edu.wpi.first.wpilibj.DataLogManager;
 import edu.wpi.first.wpilibj.RobotBase;
 import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
 import edu.wpi.first.wpilibj.simulation.ElevatorSim;
 import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
 import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
 import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Commands;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.Constants.RobotConstants;
+import frc.robot.commands.ElevatorCommands;
 import frc.robot.parameters.ElevatorLevel;
 import frc.robot.util.LimitSwitch;
 import frc.robot.util.MotorController;
 import frc.robot.util.RelativeEncoder;
 import frc.robot.util.TalonFXAdapter;
+import java.util.Set;
 
-public class Elevator extends SubsystemBase {
+@RobotPreferencesLayout(groupName = "Elevator", row = 1, column = 0, width = 1, height = 2)
+public class Elevator extends SubsystemBase implements ShuffleboardProducer {
+  @RobotPreferencesValue
+  public static RobotPreferences.BooleanValue ENABLE_TAB =
+      new RobotPreferences.BooleanValue("Elevator", "Enable Tab", false);
 
   // physical parameters of the elevator
-  private static final double GEAR_RATIO = (60.0 / 12.0) * (24.0 / 15.0);
-  private static final double SPROCKET_DIAMETER = 0.0475; // 4.75 cm
-  private static final double MASS = 2.0; // kilograms
+  private static final double GEAR_RATIO = ((60.0 / 12.0) * (24.0 / 15.0)) / 2;
+  private static final double SPROCKET_DIAMETER = 0.05; // 5 cm
+  private static final double MASS = 1.5; // kilograms
   private static final double METERS_PER_REVOLUTION = (SPROCKET_DIAMETER * Math.PI) / GEAR_RATIO;
 
+  private static final double MAX_HEIGHT = 1.42;
+  public static final double MIN_HEIGHT = 0.033; // in meters
```

Comment:
Should we define the heights relative to the floor rather than the bottom of the elevator so we can better reason about the positioning for reef levels?